libraries/AP_HAL_ChibiOS/hwdef/StellarH7V2/hwdef.dat

```diff
@@ -0,0 +1,181 @@
+# hw definition file for Stellar H7V2
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# board ID for firmware load
+APJ_BOARD_ID AP_HW_StellarH7V2
+
+# crystal frequency, setup to use external oscillator
+OSCILLATOR_HZ 8000000
+
+FLASH_SIZE_KB 2048
+env OPTIMIZE -Os
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 12
+define CH_CFG_ST_RESOLUTION 16
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 for IMU1 (MPU6000)
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PD7 SPI1_MOSI SPI1
+PC15 IMU1_CS CS
+
+# SPI2 for MAX7456
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+PB12 MAX7456_CS CS
+
+# SPI4 NAND W25M01G/2G Not supported yet
+PE12 SPI4_SCK SPI4
+PE13 SPI4_MISO SPI4
+PE14 SPI4_MOSI SPI4
+PE11 FLASH_CS CS
+
+# I2C bus
+I2C_ORDER I2C2
+
+# I2C2
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# ADC
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+PA4 BATT2_CURRENT_SENS ADC1 SCALE(1)
+
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 10
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT2_CURR_PIN 7
+define HAL_BATT_VOLT_SCALE 11.0
+define HAL_BATT_CURR_SCALE 10.0
+
+# LED
+# green LED1 marked as B/E
+# blue LED0 marked as ACT
+define AP_NOTIFY_GPIO_LED_2_ENABLED 1
+PE3 LED0 OUTPUT LOW GPIO(90)
+PE4 LED1 OUTPUT LOW GPIO(91)
+define HAL_GPIO_A_LED_PIN 91
+define HAL_GPIO_B_LED_PIN 90
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 USART6 UART7 UART8
+
+# USART1
+PA10 USART1_RX USART1
+PA9  USART1_TX USART1
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_None
+
+# USART2
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_None
+
+# USART3
+PD9 USART3_RX USART3
+PD8 USART3_TX USART3
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_None
+
+# UART4 (RCIN)
+PB9 UART4_TX UART4
+PB8 UART4_RX UART4
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_RCIN
+
+# USART6
+PC7 USART6_RX USART6
+PC6 USART6_TX USART6
+define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_GPS
+
+# UART7
+PE7 UART7_RX UART7
+PE8 UART7_TX UART7
+define DEFAULT_SERIAL7_PROTOCOL SerialProtocol_MSP_DisplayPort
+
+# UART8
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+define DEFAULT_SERIAL8_PROTOCOL SerialProtocol_ESCTelemetry
+
+# CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+PD3 GPIO_CAN1_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(70)
+
+# Motors
+PB0  TIM3_CH3  TIM3  PWM(1) BIDIR GPIO(50)
+PB1  TIM3_CH4  TIM3  PWM(2)  GPIO(51)
+PA0  TIM5_CH1  TIM5  PWM(3) BIDIR GPIO(52)
+PA1  TIM5_CH2  TIM5  PWM(4)  GPIO(53)
+PA2  TIM5_CH3  TIM5  PWM(5)  GPIO(54)
```

Comment:
You may as well add BIDIR to PWM(5) as well - there is no downside with this configuration